react/src/base/inputs/SprkInputContainer/SprkInputContainer.test.js

```diff
@@ -96,48 +100,561 @@ describe('SprkInputContainer:', () => {
     expect(bothCustom.find('label').getDOMNode().htmlFor).toBe(
       bothCustom.find('input').getDOMNode().id,
     );
+
+    const bothMatch = mount(
+      <SprkInputContainer>
+        <SprkLabel htmlFor="match">htmlFor and id Match</SprkLabel>
+        <SprkInput id="match" />
+      </SprkInputContainer>,
+    );
+    expect(bothMatch.find('input').getDOMNode().id).toBe('match');
+    expect(bothMatch.find('label').getDOMNode().htmlFor).toBe('match');
+  });
+
+  it(`should set for value of label to match
+  input id if mismatching on grandchildren`, () => {
+    const wrapper = mount(
+      <SprkInputContainer>
+        <div className="sprk-b-InputContainer__icon-container">
+          <SprkLabel htmlFor="testFor">Mismatching Values</SprkLabel>
+          <SprkInput id="testID" />
+        </div>
+      </SprkInputContainer>,
+    );
+    expect(wrapper.find('label').getDOMNode().htmlFor).toBe('testID');
+
+    const customFor = mount(
+      <SprkInputContainer>
+        <div className="sprk-b-InputContainer__icon-container">
+          <SprkLabel>With custom for</SprkLabel>
+          <SprkInput id="testID" />
+        </div>
+      </SprkInputContainer>,
+    );
+    expect(customFor.find('label').getDOMNode().htmlFor).toBe('testID');
+
+    const customID = mount(
+      <SprkInputContainer>
+        <div className="sprk-b-InputContainer__icon-container">
+          <SprkLabel htmlFor="testFor">Input has custom ID</SprkLabel>
+          <SprkInput />
+        </div>
+      </SprkInputContainer>,
+    );
+    expect(customID.find('label').getDOMNode().htmlFor).toBe(
+      customID.find('input').getDOMNode().id,
+    );
```

Comment:
I think `customFor` and `customId` are switched in these two tests. It might also make sense to split these out into a bunch of individual tests.